Bump cardano-node to (almost) 1.28.0

Anviking opened the pull request:
# Issue Number

ADP-952

# Overview

- [x] Bump to cardano-node master, in preparation for 1.28.0 release (input-output-hk/cardano-haskell#49)
- [x] Update readme 

# Comments

Basic Alonzo Era support is a non-goal for this PR. I implemented `fromAlonzoTx` and `fromAlonzoBlock` as `error "unimplemented"`s for now. Will fill them in in a follow-up PR, perhaps with integration cluster support at the same time.


- [Hydra jobset](https://hydra.iohk.io/jobset/Cardano/cardano-wallet-pr-2755#tabs-evaluations)
- [Cabal build](https://buildkite.com/input-output-hk/cardano-wallet-nightly/builds?branch=anviking%2FADP-952%2Fcardano-node-1.28.0)

Anviking committed:
Bump cardano-node to 1.28.0-pre

- Updates for cardano-node 1.28.0
- local-cluster: Add alonzo genesis
- Regenerate nix
- swagger: Add alonzo era to ApiNetworkParameters
- Api.Types: Update golden JSON files with Alonzo era info
- Bump iohk-nix to latest master branch
- Update readme

Co-Authored-By: Rodney Lorrimar <rodney.lorrimar@iohk.io>

Anviking commented on lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs:
```diff
@@ -1191,36 +1207,42 @@ mkUnsignedTx era ttl cs md wdrls certs fees =
         ShelleyBasedEraShelley -> toShelleyTxOut
         ShelleyBasedEraAllegra -> toAllegraTxOut
         ShelleyBasedEraMary -> toMaryTxOut
+        ShelleyBasedEraAlonzo -> error "todo: toAlonzoTxOut"
```
```suggestion
        ShelleyBasedEraAlonzo -> error "todo(ADP-952): toAlonzoTxOut"
```

rvl replied:
If you use something like this:
```haskell
error "toAlonzoTxOut unimplemented" -- TODO: [ADP-952] toAlonzoTxOut
```
then it's easy to find all your TODOs with lentil:
```shell
$ lentil -t ADP-952 lib
263 source files [****************] 100%
lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs
   521  fromAlonzoBlock [ADP-952]
   923  fromAlonzoTx [ADP-952]
  1148  datumHash [ADP-952]
  1167  datumHash [ADP-952]
  1186  datumHash [ADP-952]
 
lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs
  1173  We presumably need to provide the protocol params if our tx uses
        scripts? [ADP-952]
  1220  toAlonzoTxOut [ADP-952]
```

Anviking committed:
Remove readWalletMeta from restoreBlocks

`readWalletMeta` has a hidden dependency on un up-to-date TimeInterpreter.
Under heavy load, on the local cluster with short horizons, it seems to
occasionally not be up to date.

The removed log message had Debug severity, so keeping it shouldn't be
that important.

Outstanding problems:
- Technically the pool follower also depends on an up-to-date
TimeInterpreter. Hopefully problems here are less likely than for the
pool worker.
- `readWalletMeta` should either never fail with PastHorizon errors, or
make the possibility clear to followers.

Anviking committed:
Use -j 2 for macOS hydra integration tests
